src/attention.jl

```diff
@@ -0,0 +1,125 @@
+const AA3{T} = AbstractArray{T,3}
+const AA4{T} = AbstractArray{T,4}
+const AA{N,T} = AbstractArray{T,N}
+
+"""
+    dot_product_attention(query, key, value; [bias, droput_fn, mask, num_heads])
```

Comment:
Bikeshedding, but can we have simpler keyword names? I think the ideal is one word 3-7 chars, no underscores, no single letters. 
```suggestion
    dot_product_attention(query, key, value; [bias::Bool, dropout::Function, mask::Bool, heads::Int])
```
Not so sure types help.